perf/smallbank_workload/src/smallbank_tranformer.rs

```diff
@@ -28,20 +31,76 @@ use sawtooth_sdk::signing;
 
 use playlist::bytes_to_hex_str;
 use playlist::make_addresses;
-use smallbank::SmallbankTransactionPayload;
+use smallbank::{SmallbankTransactionPayload, SmallbankTransactionPayload_PayloadType};
 
 /// Transforms SmallbankTransactionPayloads into Sawtooth Transactions.
 pub struct SBPayloadTransformer<'a> {
     signer: &'a signing::Signer<'a>,
+    dependencies: DependencyTracker<u32>,
 }
 
 impl<'a> SBPayloadTransformer<'a> {
     pub fn new(signer: &'a signing::Signer) -> Self {
-        SBPayloadTransformer { signer: signer }
+        SBPayloadTransformer {
+            signer: signer,
+            dependencies: DependencyTracker::new(),
+        }
+    }
+
+    fn add_signature_if_create_account(
+        &mut self,
+        payload: &SmallbankTransactionPayload,
+        signature: String,
+    ) {
+        if payload.get_payload_type() == SmallbankTransactionPayload_PayloadType::CREATE_ACCOUNT {
+            self.dependencies
+                .add_signature_for_id(payload.get_create_account().get_customer_id(), signature);
+        }
```

Comment:
It doesn't seem like this needs to be a separate function. It just adds an extra layer of indirection.

Reply:
It names the piece of code with a description of what it does. Without it there would be no explanation for what that piece of code did. There are certainly pieces of code here that don't have a name, and could, but it is moderately helpful to provide a name for this since it isn't directly useful for making this transaction.

Reply:
Hmm, that is a a fair point. This may just be a matter of taste. _In my opinion_, the if block is straightforward. Since the code doesn't need to be reused, I see moving the code behind a function as making the code more difficult to understand because you have to track down the function. I think a comment would be just as helpful to clarify what is going on.

```
+.       // Adds a dependency for the customer id if this is a create account transaction
+        if payload.get_payload_type() == SmallbankTransactionPayload_PayloadType::CREATE_ACCOUNT {
+            self.dependencies
+                .add_dependency(payload.get_create_account().get_customer_id(), signature);
+        }
```

That being said, looking at my reasons for indirection below, this function does allow you to maintain the same level of abstraction in the function where it is used.

Also, with whatever we decide on for the dependency tracker method names, this function should follow suit.

Reply:
I think I am going to call the DependencyTracker, SignatureTracker but have the instance still be dependencies. So other uses on the Transformer class will refer to dependency.